Hydro Mini-Overhaul and Buffs: God I love gardening.

courierbravo opened the pull request:
- Changed the layout of hydroponics, the garden, and the southern end of the library to create a new hydroponics experience.
- Added a new subtype of the nutrient vendor, the HydroVend. Basically, just the nutrivend but with tools in it.
- Added a new subtype of the nutrient vendor, the GardenVend. Its like the nutrivend, but you have to pay for it. Has some new items included, too!
- Added a storefront to hydroponics. Go get your groceries!
- Added six new magazines: Horticulture Today! issues one through six.
- Added a bunch of new random item spawners.
- Added back access to the command stairwell, to make it an actually usable emergency exit.
- Tweaked the blue color for the hydroponics overalls and apron, making it a less intense blue. Also made it a defined color instead of an undefined hex code.
- Shamelessly buffed the biogenerator by reducing the time it takes to make products by half. Also, reduced the cost of several items, and added cost to items that did not have one defined.
- Buffed the xenobotany vendor, giving it more items to vend and more fertilizer.
- Added six new sprites for the horticulture magazines.

courierbravo commented:
!wip

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/0f09c0a5-d9f2-4f8e-aab0-a9659d1f63a2)
missing comma here

should be:
```
/obj/item/reagent_containers/glass/bottle/ammonia = 10,
/obj/item/reagent_containers/glass/bottle/diethylamine = 5,
/obj/random/horticulture = 4
```

courierbravo commented:
fixed the missing comma, lets see what else breaks lol

courierbravo commented:
!review

Ben10083 commented:
Noticing that public and hydroponics do not appear to have a biogenerator anymore (the grey disposal bin), intentional?

courierbravo commented:
Its there. The biogenerator got moved to deck 1 for hydro, and the mini just has a broken sprite that idk how to fix.

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/924633b6-0249-433e-a6f6-a9a13b00869c)
This looks worse than before, please revert this change. Random horizon mapping changes should be out of scope of a garden PR anyways.

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/3dcfa878-2fcb-4859-9b1c-8ff1279a27c2)
there should be a seed extractor here, ideally right by seed storage vendor, so that hydroponicists don't have to go up and down the stairs again and again to extract seeds

some less frequently used machinery or gear can be on the lower level (like honey extractor or whatever), but the main working area for hydroponics is going to be the upper level still

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/3daf3c38-cb9f-4a8a-b9aa-367ef3d694fc)
I do not think these shutters should start closed
cooks need that to cook food, and currently hydroponicists are pretty rare to provide that consistently

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/239d8172-1cff-48ff-b49f-0b8dec6f68f3)
engineering floor markings should use the engineering color
this new "warm yellow" color is closer to the brown floor decals color used in operations, than to the brighter yellow floor decals color used in engineering

DreamySkrell commented on code/game/objects/random/misc.dm:
```diff
@@ -530,3 +530,30 @@
+		/obj/item/yarn/yellow
+	)
+
+/obj/random/horticulture
```
change the type path to `/obj/random/magazine/horticulture`

DreamySkrell commented on code/game/objects/random/biogenerator_randoms.dm:
```diff
@@ -0,0 +1,82 @@
+// This list of randoms is intended for use in the biogenerator.
+// These ones are sanitized of anything funky compared to other random item lists
+
+/obj/random/biogenerator/backpack
+	name = "random backpack"
+	icon = 'icons/obj/storage/backpack.dmi'
+	icon_state = "backpack"
+	spawnlist = list(
+		/obj/item/storage/backpack,
+		/obj/item/storage/backpack/medic,
+		/obj/item/storage/backpack/security,
+		/obj/item/storage/backpack/industrial,
+		/obj/item/storage/backpack/hydroponics,
+		/obj/item/storage/backpack/pharmacy,
+		/obj/item/storage/backpack/satchel,
+		/obj/item/storage/backpack/toxins,
+		/obj/item/storage/backpack/satchel/leather,
+		/obj/item/storage/backpack/satchel/eng,
+		/obj/item/storage/backpack/satchel/med,
```
This being a random list is fine, but it's like, way too expansive and random.
It should only include plain backpacks/satchels/duffelbags/messengerbags/etc.
Plain as in, not departmental.

Both cause the biogenerator can be used by offships and the likes.
And also cause it'd require generating a lot of useless backpacks if you want to give someone a generic one.

DreamySkrell commented on code/game/objects/random/biogenerator_randoms.dm:
```diff
@@ -0,0 +1,82 @@
+		/obj/item/storage/backpack/messenger/sec,
+		/obj/item/storage/backpack/messenger/tox
+	)
+/obj/random/biogenerator/gloves
+	name = "random gloves"
+	icon = 'icons/obj/clothing/gloves.dmi'
+	icon_state = "yellow"
+	spawnlist = list(
+		/obj/item/clothing/gloves/black,
+		/obj/item/clothing/gloves/black/tajara,
+		/obj/item/clothing/gloves/black/unathi,
+		/obj/item/clothing/gloves/black_leather,
+		/obj/item/clothing/gloves/black_leather/tajara,
+		/obj/item/clothing/gloves/black_leather/unathi,
```
Same comment as for the random backpacks.
I think this list is way too big and random to be useful.

DreamySkrell commented on code/game/machinery/biogenerator.dm:
```diff
@@ -55,6 +57,26 @@
 	object = /obj/item/reagent_containers/food/snacks/soywafers
 	cost = 150
 
+/singleton/biorecipe/food/mre
+	name = "Meals Ready to Eat"
+	object = /obj/random/mre
+	cost = 450
+
```
I think the biogenerator should only generate food ingredients or whatever, stuff that can be used to make food.
Not like, actual food that is ready to eat from the start, like MREs.

DreamySkrell commented on code/game/machinery/biogenerator.dm:
```diff
@@ -126,66 +158,163 @@ ITEMS
+	object = /obj/item/storage/box/goldstar
 	cost = 600
 
+/singleton/biorecipe/item/googly
+	name = "Box of Googly Eyes"
+	object = /obj/item/storage/box/googly
+	cost = 600
```
please no
I don't want to see people spamming googly eyes on everything every round

DreamySkrell commented on code/game/machinery/vending_types.dm:
```diff
@@ -867,27 +867,104 @@
 		/obj/item/reagent_containers/syringe = 5,
 		/obj/item/storage/bag/plants = 5
-	)
+		)
```
fix formatting on these closing parentheses - should be as it was before

how it is:
```
= list(
    /foo
    )
```

how it should be:
```
= list(
    /foo
)

DreamySkrell commented on code/game/objects/items/toys.dm:
```diff
@@ -395,7 +434,6 @@
 	promote Dominian arts among their foreign partners and to proselytize their state religion to eager masses."
 	icon_state = "witchfindernovel"
 	item_state = "witchfindernovel"
-
```
readd this newline

DreamySkrell commented:
![image](https://github.com/user-attachments/assets/176506d3-f309-446f-bb1c-b68f44a19b65)
the `/obj/effect/map_effect/door_helper/unres` arrow on this door means hydroponicists can enter the kitchen freely, but can't get back the same way back to hydroponics

DreamySkrell commented:
Other than the requested changes, the PR is a good and positive addition.

courierbravo commented:
To address first to last:
-I dont know how or why that change happened because I certainly did not make it. Possible issue with mapmerge tool? If i change something i detail it. I explicitly avoid making any change i dont at least acknowledge
-Seed extractor will be moved up.
-A frequent thing I see chefs doing is breaking into hydro to grow their own stuff. I don't think this is acceptable behavior oocly as its very blatantly just doing another role's job. Thus discouraging anyone from playing the role. I want to keep the hard division due to this. If a chef really wants a specific produce, the garden isn't very far away at all. 
-Color was changed to match the maintenance door secondary color, which I think is a bit more fitting. Engineering yellow's defined hex color is pretty harsh and some people have complained about it to me. I'll revert for now, but this is a problem I'll be trying to solve at a different point.
-Path change is smart, will do that.
-Regarding backpacks and gloves. I created such a large pool to try and give options for things to sell at the front stand. You're right that its a broad list, but breaking it down to very generic items feel like its breaking that. Likewise, its not like offships dont have access to corporate themed backpacks through several means. Trash piles, crates, random spawns, etc. So, I don't think its really an issue from that aspect. Would a middle ground be to split into "generic" and "random"? 
-The biogenerator already makes rations and soywafers, both food items ready to eat. Though considered _bad_ now, they used to be seen as more normal food items in older times. Can we substitute with a more on brand MRE? Something like a custom one I could add that has on theme food. Tofu based foodstuffs and such. 
-I'd like to keep the googly eyes. Whats the point in having fun items if we don't get to use them? We added stickers, and no one ever does anything with them. I can go take photos of some of the things in my own irl workplace we've stuck googly eyes to, if you'd like.
-Will fix formatting
-Will add back line
-I could add back access to the other kitchen doors, which would fix that.

courierbravo commented:
-fixed toys
-fixed vending indents
-placed second seed extractor to up top
-made both kitchen door and hydro door share access lists for chef and hydro. So both can access some of each others areas if required.
-i think i fixed the change to the window panes on deck 3? I seriously have no clue why they're like that.
-altered bottom level to make it more reasonable for the kitchen to access. While I'm still against the idea that the kitchen should have any access to hydroponics at all, the facts of the matter compels me to make this change. 
-reverted flooring decal changes
-divides biogen randoms into true random and generic items. true randoms still contain the long list of backpacks and gloves (sans generic options)

To do:
-Defend googly eyes with my life
-MRE code

courierbravo commented:
note: on backburner due to life, work will resume on 9/01/24 or 9/02/24

courierbravo commented:
Removed the MRE and google eyes from biogenerator
